flor/journal/file.py

```diff
@@ -1,13 +1,21 @@
 from .entry import DataVal, DataRef, Bracket, EOF, make_entry
 
 from flor import shelf
+from ..logger import Logger
 
 import json
 import pathlib
 from typing import Union, List
 
 entries: List[Union[DataRef, DataVal, Bracket, EOF]] = []
 
+def _proc_log_record(log_record):
+    if isinstance(log_record, DataRef):
+        log_record.set_ref_and_dump(shelf.get_pkl_ref())
+    return json.dumps(log_record.jsonify()) + pathlib.os.linesep
+
+entries_logger = Logger()
+entries_logger.register_pre(_proc_log_record)
```

Comment:
I think I see what you're doing here. You're being clear about the code that gets run on `pre`.  But the user won't be changing the Flor code base, so this show-and-tell format is better off in a readme or Docs.

Reply:
I'm not too sure what you mean by "show-and-tell format." It's registering a preprocessing step particular to `entries_logger` but not all loggers.